MultiSource/Benchmarks/Ptrdist/anagram/anagram.c

```diff
@@ -440,14 +547,43 @@ void BuildWord(char * pchWord) {
 /* AddWords -- build the list of candidates */
 void
 AddWords(void) {
-    char * pch = pchDictionary;     /* walk through the dictionary */
+    // CHECKEDC : declared bounds at variable declaration
+    // FIXME
+    // Since it has function call within body, it is not proper usage
+    // that global variable is directly used for bounds declaration
+    // In another function call, it can modify those global variables
+    // To prevent this problem, add additional pointer variable to hold bounds
+    _Array_ptr<char> pchLowerBounds = pchDictionary;
+    _Array_ptr<char> pchUpperBounds = pchDictionary + pchDictionarySize;
+    _Array_ptr<char> pch : bounds(pchLowerBounds, pchUpperBounds) = pchDictionary;     /* walk through the dictionary */
 
     cpwCand = 0;
 
+    // CHECKEDC : memory read, dereference(in-bounds check)
+    // dynamic_check(pch >= pchLowerBounds && pch < pchUpperBounds)
     while (*pch) {
+        // 1. find out each word in dictionary
+        // 2. if word letter size is beyond phrase size, skip word
+        // 3. check if frequency(word letter) < frequency(phrase letter) for all letters of word
+        // 4. add words as candidates for anagram for this phrase
         if ((pch[1] >= cchMinLength && pch[1]+cchMinLength <= cchPhraseLength)
             || pch[1] == cchPhraseLength)
-	    BuildWord(pch+2);
+            // pch[0] = length of word + 3 bytes(length of words, number of letter, NULL string)
+            // pch[1] = number of letters
+            // pch[2..] = word itself
+        {
+            // FIXME
+            // - add addtional variable to represent words bounds (wordStart/wordEnd)
+            // - max word length is encoded in pch[0], use it
+            // - rewrite function call to hold bounds start/end in function argument
+            unsigned char wordLength = pch[0];
+            _Array_ptr<char> wordStart = pch;
+            _Array_ptr<char> wordEnd = pch+wordLength;
```

Comment:
Let me explain why I passed wordStart/wordEnd to buildWord instead of pchLower/UpperBound.
pchLower/UpperBounds is dictionary start/end but wordStart/End is bounds for a word not dictionary
In AddWords function, it iterates all words in dictionary
Therefore, pch pointer advances from pchLowerBounds to pchUpperBounds to traverse whole words in dictionary
In BuildWord function, it processes each word
So, I have passed bounds about wordStart/wordEnd for buildWord
It is the reason why I have used wordStart/End as bounds
To represent precise bounds information, I have passed it as argument for BuildWord function

This relationship (pchLowerBounds <= wordStart < wordEnd < pchUpperBounds) works